src/librustc_mir/dataflow/impls/borrows.rs

```diff
@@ -231,21 +268,17 @@ impl<'a, 'gcx, 'tcx> BitDenotation for Borrows<'a, 'gcx, 'tcx> {
             panic!("could not find statement at location {:?}");
         });
 
+        debug!("Borrows::statement_effect: stmt={:?}", stmt);
         match stmt.kind {
             mir::StatementKind::Assign(ref lhs, ref rhs) => {
                 // Make sure there are no remaining borrows for variables
                 // that are assigned over.
-                if let Place::Local(ref local) = *lhs {
-                    // FIXME: Handle the case in which we're assigning over
-                    // a projection (`foo.bar`).
-                    self.kill_borrows_on_local(sets, local);
-                }
+                self.kill_borrows_on_place(sets, location, lhs);
```

Comment:
There is a NOTE  in a comment below this line, and I always wonder whether the directive in this comment has been (or is being) followed... 

The local `assigned_place` has presumably been alpha-renamed, probably to `lhs` here)... but in any case, do we need to make modifications to `propagate_call_return`  to make the logic here match up with the logic there? Are the comments still relevant in the first place?

Reply:
Okay, after thinking the matter through, I think the PR as it stands is sound; it is just more conservative than it could be, because I *think* we could apply the same place-overwritten borrow-killing logic to `propagate_call_return`, since that represents an overwrite of the `dest_place` that is passed to that method.

It would be best to first come up with a separate example illustrating this, rather than blindly adding the code to the PR.